docs/modules/ROOT/pages/how-tos/configuring/prefetching-dependencies.adoc

```diff
@@ -298,6 +302,58 @@ spec:
 * From the {ProductName} *Applications* view, go to *Activity > Latest commits*.
 //include::../partials/con_hermetic_verification.adoc[]
 
+== [[yarn]]Enabling prefetch builds for `yarn`
+
+Supported versions: 1.x and 3.x. Cachi2 automatically detects the version of `yarn` and fetches any dependencies you declare in your `package.json` and `yarn.lock` project files.
+
+.Prerequisites
+
+* You have an up-to-date `yarn.lock` file in your source repository. To make sure that you have the latest `yarn.lock` file, or to create a lockfile, run the `yarn install` command. If `yarn.lock` is not up-to-date, Cachi2 will not fetch the dependencies.
```

Comment:
Will Cachi2 not fetch any dependencies if the lockfiles are not up to date? Or is the behavior just that the locked dependencies will be fetched (and therefore your desired latest dependencies might not be fetched properly)?

(the comment is relevant for other package managers too)

Reply:
**TL;DR** It will fetch for `bundler` but not for `yarn`. I have to fix it...

It depends on the implementation details of each package manager.  For example for `yarn`, we use native mechanisms to pre-fetch the dependencies, the lockfile must be up to date. We use the command:  `yarn install --frozen-lockfile ...`  which fails in that case. For other package managers like `pip`, we just manually parse `requirements-txt` so it does not make sense to do any validation there.